### PR TITLE
Add projection option:  update lock threshold

### PR DESCRIPTION
**Pull request by codeliner**
This PR provides the implementation for the new projection option `update lock threshold`, see https://github.com/prooph/event-store/pull/331

### CODE REVIEW

File: src/Projection/MariaDbProjectionManager.php [hunk 1]
```diff
@@ -107,7 +108,8 @@ public function createReadModelProjection(
             $options[PdoEventStoreReadModelProjector::OPTION_LOCK_TIMEOUT_MS] ?? PdoEventStoreReadModelProjector::DEFAULT_LOCK_TIMEOUT_MS,
             $options[PdoEventStoreReadModelProjector::OPTION_PERSIST_BLOCK_SIZE] ?? PdoEventStoreReadModelProjector::DEFAULT_PERSIST_BLOCK_SIZE,
             $options[PdoEventStoreReadModelProjector::OPTION_SLEEP] ?? PdoEventStoreReadModelProjector::DEFAULT_SLEEP,
-            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH
+            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH,
+            $options[PdoEventStoreProjector::OPTION_UPDATE_LOCK_THRESHOLD] ?? PdoEventStoreProjector::DEFAULT_UPDATE_LOCK_THRESHOLD
```
Comment (prolic) [inline src/Projection/MariaDbProjectionManager.php]: Should be read model

File: src/Projection/MySqlProjectionManager.php [hunk 2]
```diff
@@ -107,7 +108,8 @@ public function createReadModelProjection(
             $options[PdoEventStoreReadModelProjector::OPTION_LOCK_TIMEOUT_MS] ?? PdoEventStoreReadModelProjector::DEFAULT_LOCK_TIMEOUT_MS,
             $options[PdoEventStoreReadModelProjector::OPTION_PERSIST_BLOCK_SIZE] ?? PdoEventStoreReadModelProjector::DEFAULT_PERSIST_BLOCK_SIZE,
             $options[PdoEventStoreReadModelProjector::OPTION_SLEEP] ?? PdoEventStoreReadModelProjector::DEFAULT_SLEEP,
-            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH
+            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH,
+            $options[PdoEventStoreProjector::OPTION_UPDATE_LOCK_THRESHOLD] ?? PdoEventStoreProjector::DEFAULT_UPDATE_LOCK_THRESHOLD
```
Comment (prolic) [inline src/Projection/MySqlProjectionManager.php]: Read model again

File: src/Projection/PostgresProjectionManager.php [hunk 3]
```diff
@@ -107,7 +108,8 @@ public function createReadModelProjection(
             $options[PdoEventStoreReadModelProjector::OPTION_LOCK_TIMEOUT_MS] ?? PdoEventStoreReadModelProjector::DEFAULT_LOCK_TIMEOUT_MS,
             $options[PdoEventStoreReadModelProjector::OPTION_PERSIST_BLOCK_SIZE] ?? PdoEventStoreReadModelProjector::DEFAULT_PERSIST_BLOCK_SIZE,
             $options[PdoEventStoreReadModelProjector::OPTION_SLEEP] ?? PdoEventStoreReadModelProjector::DEFAULT_SLEEP,
-            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH
+            $options[PdoEventStoreReadModelProjector::OPTION_PCNTL_DISPATCH] ?? PdoEventStoreReadModelProjector::DEFAULT_PCNTL_DISPATCH,
+            $options[PdoEventStoreProjector::OPTION_UPDATE_LOCK_THRESHOLD] ?? PdoEventStoreProjector::DEFAULT_UPDATE_LOCK_THRESHOLD
```
Comment (prolic) [inline src/Projection/PostgresProjectionManager.php]: Read model

File: src/Projection/PdoEventStoreProjector.php [hunk 4]
```diff
@@ -930,6 +949,17 @@ private function createLockUntilString(DateTimeImmutable $from): string
         return $from->modify('+' . $secs .' seconds')->format('Y-m-d\TH:i:s') . '.' . $resultMicros;
     }
 
+    private function shouldUpdateLock(DateTimeImmutable $now): bool
```
Comment (prolic) [inline src/Projection/PdoEventStoreProjector.php]: We need this on read model Projection as well

Comment (codeliner) [reply src/Projection/PdoEventStoreProjector.php]: oh of course, good catch! Totally missed that.

Comment (codeliner) [reply src/Projection/PdoEventStoreProjector.php]: done

File: src/Projection/MariaDbProjectionManager.php (hunk 1 above)
Comment (codeliner) [reply src/Projection/MariaDbProjectionManager.php]: copy & paste is not my best skill :D

**Comment by codeliner** [comment]
Tests are scheduled for tomorrow.

**Comment by codeliner** [comment]
@prolic tests are working now. My threshold check was wrong, see last commit. If PR is accepted, I'd release a new minor event-store version, pin pdo-event-store to that new version, merge this PR and release a new minor pdo-event-store version, too

**Comment by codeliner** [comment]
Going to add a few more tests for `shouldUpdateLock` using reflection. Just to be sure that the method works correctly in all scenarios.

**Comment by codeliner** [comment]
Ok this PR is ready now. I've added a few more tests, event-store version is set and the threshold integration test works with increased sleep time due to failing travis tests caused by slow projection process start up

**Pull request merged**